code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -149,77 +268,118 @@
 	M.status_flags |= GOTTAGOREALLYFAST
-	M.Jitter(2)
-	M.adjustBrainLoss(0.25)
+	M.AdjustParalysis(-2.5, 0)
+	M.AdjustStunned(-2.5, 0)
+	M.AdjustWeakened(-2.5, 0, 0)
+	M.adjustStaminaLoss(-2, 0)
+	M.Jitter(5)
+	if(prob(50))
+		M.adjustBrainLoss(1)
+	M.drowsyness = max(M.drowsyness-10, 0)
 	if(prob(5))
 		M.emote(pick("twitch", "shiver"))
 	..()
 	. = 1
 
 /datum/reagent/drug/methamphetamine/overdose_process(mob/living/M)
-	if(M.canmove && !istype(M.loc, /atom/movable))
-		for(var/i = 0, i < 4, i++)
-			step(M, pick(cardinal))
-	if(prob(20))
-		M.emote("laugh")
-	if(prob(33))
-		M.visible_message("<span class='danger'>[M]'s hands flip out and flail everywhere!</span>")
-		var/obj/item/I = M.get_active_hand()
-		if(I)
-			M.drop_item()
+	if(prob(75))
+		return
+	var/effect = rand(1,7)
+	switch(volume)
+		if(1 to 59)
+			switch(effect)
+				if(1 to 3)
+					M.visible_message("<span class = 'danger'><b>[M.name]</b> can't seem to control their legs!</span>") // i do meth
+					M.Weaken(4) // so i can work longer
+					M.confused += 20 // so i can earn more
+				if(4 to 6) // so i can do more meth
+					M.visible_message("<span class = 'danger'><b>[M.name]'s</b> hands flip out and flail everywhere!</span>") // i do meth
+					M.drop_item() // so i can work longer
+				if(7) // so i can earn more
+					M.emote("laugh") // so i can do more meth
+		if(60 to INFINITY)
+			M.reagents.add_reagent("triplemeth", 10)
+			switch(effect)
+				if(1 to 3)
+					M.visible_message("<span class = 'danger'><b>[M.name]'s</b> hands flip out and flail everywhere!</span>")
+					M.drop_item()
+				if(4 to 6)
+					M.visible_message("<span class = 'danger'><b>[M.name]</b> falls to the floor and flails uncontrollably!</span>")
+					M.Weaken(10)
+					M.Jitter(10)
+				if(7)
+					M.emote("laugh")
 	..()
-	M.adjustToxLoss(1, 0)
-	M.adjustBrainLoss(pick(0.5, 0.6, 0.7, 0.8, 0.9, 1))
 	. = 1
 
-/datum/reagent/drug/methamphetamine/addiction_act_stage1(mob/living/M)
-	M.Jitter(5)
-	if(prob(20))
-		M.emote(pick("twitch","drool","moan"))
-	..()
+/datum/reagent/drug/triplemeth
+	name = "Triple Meth"
+	id = "triplemeth"
+	description = "Holy shit."
+	reagent_state = LIQUID
+	color = rgb(250,250,250)
+	overdose_threshold = 20
+	metabolization_rate = 0.2 * REAGENTS_METABOLISM
 
-/datum/reagent/drug/methamphetamine/addiction_act_stage2(mob/living/M)
-	M.Jitter(10)
-	M.Dizzy(10)
-	if(prob(30))
-		M.emote(pick("twitch","drool","moan"))
+/datum/reagent/drug/triplemeth/on_mob_life(mob/living/M)
+	M.AdjustParalysis(-M.paralysis, 0)
+	M.AdjustStunned(-M.stunned, 0)
+	M.AdjustWeakened(-M.weakened, 0, 0)
+	M.adjustStaminaLoss(-M.getStaminaLoss(), 0)
+	M.Jitter(5)
+	M.Dizzy(5)
+	M.confused += 15
+	M.adjustBrainLoss(1)
+	M.overlay_fullscreen("triplemeth1", /obj/screen/fullscreen/triplemeth1)
+	M.overlay_fullscreen("triplemeth2", /obj/screen/fullscreen/triplemeth2)
+	M.overlay_fullscreen("triplemeth3", /obj/screen/fullscreen/triplemeth3)
+	if(prob(50))
+		M.emote(pick("twitch", "shiver"))
```

Comment:
I'd recommend this just giving you the flags that hulks have---for full stun immunity (what the drug does), then having it so those flags are cleared when the reagent is deleted.

This is an amazingly powerful anti-stun drug, but its causes so many negative effects that it's barely used.